app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/item/VaultItemSshKeyContent.kt

```diff
@@ -0,0 +1,129 @@
+package com.x8bit.bitwarden.ui.vault.feature.item
+
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.navigationBarsPadding
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import com.x8bit.bitwarden.R
+import com.x8bit.bitwarden.ui.platform.components.field.BitwardenPasswordField
+import com.x8bit.bitwarden.ui.platform.components.field.BitwardenTextField
+import com.x8bit.bitwarden.ui.platform.components.header.BitwardenListHeaderText
+import com.x8bit.bitwarden.ui.vault.feature.item.handlers.VaultSshKeyItemTypeHandlers
+
+/**
+ * The top level content UI state for the [VaultItemScreen] when viewing a SSH key cipher.
+ */
+@Suppress("LongMethod")
+@Composable
+fun VaultItemSshKeyContent(
+    commonState: VaultItemState.ViewState.Content.Common,
+    sshKeyItemState: VaultItemState.ViewState.Content.ItemType.SshKey,
+    vaultSshKeyItemTypeHandlers: VaultSshKeyItemTypeHandlers,
+    modifier: Modifier = Modifier,
+) {
+    LazyColumn(modifier = modifier) {
+        item {
+            BitwardenListHeaderText(
+                label = stringResource(id = R.string.item_information),
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(horizontal = 16.dp),
+            )
+        }
+
+        item {
+            Spacer(modifier = Modifier.height(8.dp))
+            BitwardenTextField(
+                label = stringResource(id = R.string.name),
+                value = commonState.name,
+                onValueChange = { },
+                readOnly = true,
+                singleLine = false,
+                modifier = Modifier
+                    .testTag("SshKeyItemNameEntry")
+                    .fillMaxWidth()
+                    .padding(horizontal = 16.dp),
+            )
+        }
+
+        sshKeyItemState.publicKey?.let { publicKey ->
```

Comment:
Note: On `clients` (https://github.com/bitwarden/clients/pull/10825) we are not hiding publicKey/fingerprint, only private key. The former 2 are not considered private, so, these probably should be BitwardenTextField?

Reply:
Not too early at all! Thank you for taking a look and calling this out. I'll update it accordingly.